android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMTextView.java

```diff
@@ -179,11 +187,7 @@ public void onWindowFocusChanged(boolean hasWindowFocus) {
 
       if (activeView != null && activeView.equals(this)) {
         if (KMManager.InAppKeyboardLoaded) {
-          KMTextView textView = (KMTextView) activeView;
-          int selStart = textView.getSelectionStart();
-          int selEnd = textView.getSelectionEnd();
-          KMManager.updateText(KeyboardType.KEYBOARD_TYPE_INAPP, textView.getText().toString());
-          KMManager.updateSelectionRange(KeyboardType.KEYBOARD_TYPE_INAPP, selStart, selEnd);
+          updateTextContext();
```

Comment:
This is a slight change in functionality here -- do you need to do any tests to validate the change?

Reply:
Here's what I did to check the context is (now) reset

1. Install basic_kbdfr (it uses deadkeys)
2. With French Basic keyboard selected:
3. Type the deadkey <kbd>^</kbd>
4. Type <kbd>a</kbd> and verify output `â `
5. Now type another deadkey <kbd>^</kbd>
6. Hit the Android square task button, then focus back on Keyman
7. Type <kbd>a</kbd> and verify only 'a' is output (deadkey cleared)

Reply:
Is that behaviour different to the previous version?

Reply:
Previous version didn't reset context. Step 7 typing <kbd>a</kbd> would output `â`

Reply:
Excellent, that's what I was hoping!